azkaban-common/src/main/java/azkaban/project/FlowLoaderUtils.java

```diff
@@ -15,22 +15,82 @@
 */
 package azkaban.project;
 
+import azkaban.Constants;
 import azkaban.flow.CommonJobProperties;
 import azkaban.flow.Flow;
 import azkaban.project.validator.ValidationReport;
 import azkaban.utils.Props;
 import java.io.File;
 import java.io.FileFilter;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Utils to help load flows.
  */
 public class FlowLoaderUtils {
 
+  private static final Logger logger = LoggerFactory.getLogger(FlowLoaderUtils.class);
+
+  /**
+   * Gets flow or job props from flow yaml file.
+   *
+   * @param path the flow or job path delimited by ":", e.g. "flow:subflow1:subflow2:job3"
+   * @param flowFile the flow yaml file
+   * @return the props from yaml file
+   */
+  public static Props getPropsFromYamlFile(final String path, final File flowFile) {
```

Comment:
What do you think of passing nodeBean inside instead of the original file? Otherwise, If one day we want to implement getFlowTriggerFromYamlFile, it will need to parse the yaml file again.

Reply:
Will getFlowTrigger be called together with getProps? If not, then we need to store the parsed nodeBean somewhere if you want to reuse it. But this might not work well if someone overrides the yaml file. I think reading the yaml file from DB and parsing the file is necessary since the file could potentially change. Also I think parsing the yaml file to NodeBean is straightforward and shouldn't be costly. Correct me if I'm wrong.

Reply:
'''
Will getFlowTrigger be called together with getProps
'''
No

Reply:
If it's really needed for optimization in the future, we can split this method into two parts. But again the main concern is the flow file itself might change, it is better to read from DB directly instead of caching the parsed value in nodeBean.